Phalcon/Db/Adapter/Pdo/Sqlsrv.php

```diff
@@ -344,71 +342,56 @@ public function query($sqlStatement, $bindParams = null, $bindTypes = null)
     }
 
     /**
-     * Sends SQL statements to the database server returning the success state.
-     * Use this method only when the SQL statement sent to the server doesn't return any rows
-     * <code>
-     * //Inserting data
-     * $success = $connection->execute("INSERT INTO robots VALUES (1, 'Astro Boy')");
-     * $success = $connection->execute("INSERT INTO robots VALUES (?, ?)", array(1, 'Astro Boy'));
-     * </code>.
-     *
-     * @param string $sqlStatement
-     * @param mixed  $bindParams
-     * @param mixed  $bindTypes
+     * Creates a PDO DSN for the adapter from $this->config settings.
      *
-     * @return bool
+     * @return string
      */
-//    public function execute($sqlStatement, $bindParams = null, $bindTypes = null)
-    //    {
-    //        $eventsManager = $this->_eventsManager;
-    //
-    //        /*
-    //         * Execute the beforeQuery event if a EventsManager is available
-    //         */
-    //        if (is_object($eventsManager)) {
-    //            $this->_sqlStatement = $sqlStatement;
-    //            $this->_sqlVariables = $bindParams;
-    //            $this->_sqlBindTypes = $bindTypes;
-    //
-    //            if ($eventsManager->fire('db:beforeQuery', $this, $bindParams) === false) {
-    //                return false;
-    //            }
-    //        }
-    //
-    //        /*
-    //         * Initialize affectedRows to 0
-    //         */
-    //        $affectedRows = 0;
-    //
-    //        $pdo = $this->_pdo;
-    //
-    //        $cursor = \PDO::CURSOR_SCROLL;
-    //        if (strpos($sqlStatement, 'exec') !== false) {
-    //            $cursor = \PDO::CURSOR_FWDONLY;
-    //        }
-    //
-    //        if (is_array($bindParams)) {
-    //            $statement = $pdo->prepare($sqlStatement, array(\PDO::ATTR_CURSOR => $cursor));
-    //            if (is_object($statement)) {
-    //                $newStatement = $this->executePrepared($statement, $bindParams, $bindTypes);
-    //                $affectedRows = $newStatement->rowCount();
-    //            }
-    //        } else {
-    ////            $statement = $pdo->prepare($sqlStatement, array(\PDO::ATTR_CURSOR => $cursor));
-    ////            $statement->execute();
-    //            $affectedRows = $pdo->exec($sqlStatement);
-    //        }
-    //
-    //        /*
-    //         * Execute the afterQuery event if an EventsManager is available
-    //         */
-    //        if (is_int($affectedRows)) {
-    //            $this->_affectedRows = affectedRows;
-    //            if (is_object($eventsManager)) {
-    //                $eventsManager->fire('db:afterQuery', $this, $bindParams);
-    //            }
-    //        }
-    //
-    //        return true;
-    //    }
+    protected function getDsnDefaults(): array
```

Comment:
@DavisDS why does your `getDsnDefaults()` returns `string` instead of an `array`. All the other implementations seem to return array.

Reply:
@DavisDS the `getDsnDefaults()` method is not used at all.